doc/developer/design/20230714_show_privileges.md

```diff
@@ -0,0 +1,200 @@
+    - If specified, top level objects, such as database, schemas, and clusters are not included.
+    - If not specified then objects are limited to the current schema and top level objects are
+    - shown.
+
+The output will include the following columns:
+
+- `name: text` The name of the object.
+- `object_type: text` The type of the object.
+- `privileges: text` Human readable version of the privileges that held by the current session.
+
+Here are some example queries:
+
+```sql
+SHOW PRIVILEGES;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | table       | SELECT, INSERT
+ my_view    | view        | SELECT
+ my_cluster | cluster     | CREATE
+```
+
+```sql
+SHOW PRIVILEGES ON TABLES;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | table       | SELECT, INSERT
+```
+
+```sql
+SHOW PRIVILEGES FROM public;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | table       | SELECT, INSERT
+ my_view    | view        | SELECT
+```
+
+```sql
+SHOW PRIVILEGES ON VIEWS FROM public;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_view    | view        | SELECT
+```
+
+### `SHOW DEFAULT PRIVILEGES`
+
+Will show all default privileges
+
+The syntax will be `SHOW DEFAULT PRIVILEGES [ON <object-type>]`.
+
+- `ON <object-type>` will filter the output to only include objects of type `<object-type>`. Valid
+  values are:
+  - `TABLES` (includes views, materialized views, and sources)
+  - `TYPES`
+  - `CLUSTERS`
+  - `SECRETS`
+  - `CONNECTIONS`
+  - `DATABASES`
+  - `SCHEMAS`
+
+The output will include the following columns:
+
+- `role: text` The name of the affected role.
+- `database: text` The name of the affected database.
+- `schema: text` The name of the affected schema.
+- `object_type: text` The type of object.
+- `grantee: text` The role being granted privileges.
+- `privileges: text` Human readable version of the privileges that will be granted.
+
+Here are some example queries:
+
+```sql
+SHOW DEFAULT PRIVILEGES;
+
+ role   | database    | schema | object_type | grantee   | privileges
+--------+-------------+--------+-------------+-----------+----------------
+ PUBLIC | NULL        | NULL   | table       | dev       | INSERT, SELECT
+ joe    | materialize | NULL   | table       | PUBLIC    | SELECT
+ qa     | materialize | public | type        | scientist | USAGE
+ PUBLIC | NULL        | NULL   | cluster     | mike      | CREATE
+```
+
+```sql
+SHOW DEFAULT PRIVILEGES ON TABLES;
+
+ role   | database    | schema | object_type | grantee   | privileges
+--------+-------------+--------+-------------+-----------+----------------
+ PUBLIC | NULL        | NULL   | table       | dev       | INSERT, SELECT
+ joe    | materialize | NULL   | table       | PUBLIC    | SELECT
+```
+
+### `SHOW ROLE MEMBERSHIPS`
```

Comment:
is adding `[FOR <role-name>]` useful?

Reply:
Yes, I think that's a good idea! I'll update this.